analyzer/src/main/kotlin/managers/Bower.kt

```diff
@@ -19,20 +19,237 @@
+        }
+    }
+
+    private val stashedDirectories = HashMap<File, File>()
+
+    override fun toString(): String = PROVIDER_NAME
+
+    override fun command(workingDir: File?) = if (OS.isWindows) "bower.cmd" else "bower"
+
+    override fun prepareResolution(definitionFiles: List<File>): List<File> {
+        checkVersion(
+                Requirement.buildNPM(REQUIRED_BOWER_VERSION),
+                ignoreActualVersion = analyzerConfig.ignoreToolVersions
+        )
+        return definitionFiles
+    }
+
+    override fun resolveDependencies(definitionFile: File): ProjectAnalyzerResult? {
+        log.info { "Resolving dependencies for: '${definitionFile.absolutePath}'" }
+
+        val workingDir = definitionFile.parentFile
+        val bowerComponentsDir = File(workingDir, "bower_components")
+        var result: ProjectAnalyzerResult?
+
+        try {
+            stashDir(bowerComponentsDir)
+
+            installDependencies(workingDir)
+            val dependenciesJson = listDependencies(workingDir)
+            val rootNode = jsonMapper.readTree(dependenciesJson)
+            val packages = extractPackages(rootNode)
+            val dependenciesScope = Scope(
+                    name = SCOPE_NAME_DEPENDENCIES,
+                    dependencies = extractDependencyTree(rootNode, SCOPE_NAME_DEPENDENCIES)
+            )
+            val devDependenciesScope = Scope(
+                    name = SCOPE_NAME_DEV_DEPENDENCIES,
+                    dependencies = extractDependencyTree(rootNode, SCOPE_NAME_DEV_DEPENDENCIES)
+            )
+
+            val projectPackage = extractPackage(rootNode)
+            val project = Project(
+                    id = projectPackage.id,
+                    definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+                    declaredLicenses = projectPackage.declaredLicenses,
+                    vcs = projectPackage.vcs,
+                    vcsProcessed = processProjectVcs(workingDir, projectPackage.vcs, projectPackage.homepageUrl),
+                    homepageUrl = projectPackage.homepageUrl,
+                    scopes = sortedSetOf(dependenciesScope, devDependenciesScope)
+            )
+
+            result = ProjectAnalyzerResult(
+                    project = project,
+                    packages = packages.map { it.value.toCuratedPackage() }.toSortedSet()
+            )
+        } finally {
+            bowerComponentsDir.safeDeleteRecursively()
+            unstashDir(bowerComponentsDir)
+        }
+
+        return result
+    }
+
+    private fun stashDir(originalDir: File) {
+        if (!originalDir.isDirectory) return
+
+        val tempDir = createTempDir("analyzer", ".tmp", originalDir)
+        log.info { "Temporarily moving directory from '${originalDir.absolutePath}' to '${tempDir.absolutePath}'." }
+
+        Files.move(originalDir.toPath(), tempDir.toPath(), StandardCopyOption.ATOMIC_MOVE)
+        stashedDirectories.put(originalDir, tempDir)
```

Comment:
This `put` can be converted to an assignment.

Reply:
Done.